t/log.t

```diff
@@ -0,0 +1,514 @@
+            local ngx_log = require "ngx.log"
+            local res = ngx_log.get_errlog()
+            ngx.say("log lines:", #res / 2)
+        }
+    }
+--- log_level: info
+--- request
+GET /t
+--- response_body
+log lines:3
+--- grep_error_log eval
+qr/-->\d+/
+--- grep_error_log_out eval
+[
+"-->1
+-->2
+-->3
+",
+"-->1
+-->2
+-->3
+"
+]
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 12: filter log by level(ngx.WARN)
+--- http_config
+    lua_intercept_error_log 4m;
+--- config
+    location /t {
+        access_by_lua_block {
+            local ngx_log = require "ngx.log"
+            local status, err = ngx_log.set_errlog_filter(ngx.WARN)
+            if not status then
+                error(err)
+            end
+
+            ngx.log(ngx.INFO, "-->1")
+            ngx.log(ngx.WARN, "-->2")
+            ngx.log(ngx.ERR, "-->3")
+        }
+        content_by_lua_block {
+            local ngx_log = require "ngx.log"
+            local res = ngx_log.get_errlog()
+            ngx.say("log lines:", #res / 2)
+        }
+    }
+--- log_level: info
+--- request
+GET /t
+--- response_body
+log lines:2
+--- grep_error_log eval
+qr/-->\d+/
+--- grep_error_log_out eval
+[
+"-->1
+-->2
+-->3
+",
+"-->1
+-->2
+-->3
+"
+]
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 13: filter log by level(ngx.CRIT)
+--- http_config
+    lua_intercept_error_log 4m;
+--- log_level: info
+--- config
+    location /t {
+        access_by_lua_block {
+            local ngx_log = require "ngx.log"
+            local status, err = ngx_log.set_errlog_filter(ngx.CRIT)
+            if not status then
+                error(err)
+            end
+
+            ngx.log(ngx.INFO, "-->1")
+            ngx.log(ngx.WARN, "-->2")
+            ngx.log(ngx.ERR, "-->3")
+        }
+        content_by_lua_block {
+            local ngx_log = require "ngx.log"
+            local res = ngx_log.get_errlog()
+            ngx.say("log lines:", #res / 2)
+        }
+    }
+--- request
+GET /t
+--- response_body
+log lines:0
+--- grep_error_log eval
+qr/-->\d+/
+--- grep_error_log_out eval
+[
+"-->1
+-->2
+-->3
+",
+"-->1
+-->2
+-->3
+"
+]
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 14: set max count and reuse table
+--- http_config
+    lua_intercept_error_log 4m;
+--- config
+    location /t {
+        access_by_lua_block {
+            tab_clear = require "table.clear"
+            ngx.log(ngx.ERR, "enter 1")
+            ngx.log(ngx.ERR, "enter 22")
+            ngx.log(ngx.ERR, "enter 333")
+
+            local ngx_log = require "ngx.log"
+            local res = {}
+            local err
+            res, err = ngx_log.get_errlog(2, res)
+            if not res then
+                error("FAILED " .. err)
+            end
+            ngx.say("log lines:", #res / 2)
+
+            tab_clear(res)
+            res, err = ngx_log.get_errlog(2, res)
+            if not res then
+                error("FAILED " .. err)
+            end
+            ngx.say("log lines:", #res / 2)
```

Comment:
It seems to me that *none* of your existing test cases actually checks the log data inside the table returned by the `get_errlog()` function. All the tests simply test the size of the resulting tables, which is too loose. We should actually check the *contents* inside the table, not just the error log messages in the nginx error log file, which is a very different data flow anyway.